src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs

```diff
@@ -544,9 +546,66 @@ public static RouteHandlerBuilder MapFallback(this IEndpointRouteBuilder endpoin
                     endpointBuilder.Metadata.Add(attribute);
                 }
             }
+
+            // Add metadata provided by the delegate return type and parameter types
+            AddTypeProvidedMetadata(handler.Method, endpointBuilder.Metadata, endpoints.ServiceProvider);
```

Comment:
I thought `RequestDelegateFactory` was going to continue calling the `PopulateMetadata` methods. I think it's a strange omission to not support the `IEndpointMetadataProvider` if using the `RequestDelegateFactory` directly.

Reply:
Actually, after re-reading the conversation about this it wasn't super clear. Either way, I agree it would be better if the interfaces were supported by callers of `RequestDelegateFactory` directly so I'll take another crack at it.